main/src/main/scala/sbt/Defaults.scala

```diff
@@ -513,7 +513,8 @@ object Defaults extends BuildCommon {
     Seq(
       initialCommands :== "",
       cleanupCommands :== "",
-      asciiGraphWidth :== 40
+      asciiGraphWidth :== 40,
+      insideCI :== sys.env.get("BUILD_NUMBER").isDefined || sys.env.get("CI").isDefined
```

Comment:
Could you define this default in `globalSbtCore` rather than `configGlobal` please?